narwhal/network/src/failpoints.rs

```diff
@@ -0,0 +1,29 @@
+use anemo_tower::callback::{MakeCallbackHandler, ResponseHandler};
+use fail::fail_point;
+
+#[derive(Clone)]
+pub struct FailpointsMakeCallbackHandler {}
+
+impl FailpointsMakeCallbackHandler {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl MakeCallbackHandler for FailpointsMakeCallbackHandler {
+    type Handler = FailpointsResponseHandler;
+
+    fn make_handler(&self, _request: &anemo::Request<bytes::Bytes>) -> Self::Handler {
+        FailpointsResponseHandler {}
+    }
+}
+
+pub struct FailpointsResponseHandler {}
+
+impl ResponseHandler for FailpointsResponseHandler {
+    fn on_response(self, _response: &anemo::Response<bytes::Bytes>) {
+        fail_point!("rpc-delay");
```

Comment:
As mentioned before, if this does a thread sleep instead of an async one, then this is not going to have the intended behavior you want and could end up blocking the system and causing other unintended effects.

Reply:
I will leave a comment about this behavior but I was intending to add a tower layer to handle the tokio sleep in a future PR. Will sync with you offline about it